### PR TITLE
gh-123560: correct docs for "empty" format type for floats

### PR DESCRIPTION
**Pull request by skirpichev**
* Issue: gh-123560

----
📚 Documentation preview 📚: https://cpython-previews--123561.org.readthedocs.build/

### CODE REVIEW

**Comment by skirpichev** [comment]
CC @picnixz 

FYI, this behaviour was implemented in d3ca557150 (mostly it's a fallback code with few lines in format_float_short).

**Comment by picnixz** [comment]
I'll have a look at it tomorrow afternoon or in the week-end. I need to read your issue as well.

File: Doc/library/string.rst [hunk 1]
```diff
@@ -574,11 +574,13 @@ The available presentation types for :class:`float` and
...
-   | None    | For :class:`float` this is the same as ``'g'``, except   |
+   | None    | For :class:`float` this is like the ``'g'`` type, except |
    |         | that when fixed-point notation is used to format the     |
    |         | result, it always includes at least one digit past the   |
-   |         | decimal point. The precision used is as large as needed  |
-   |         | to represent the given value faithfully.                 |
+   |         | decimal point, thus switch to the scientific notation    |
+   |         | happens when ``exp >= p - 1``.  If precision is not      |
+   |         | specified, the precision used is as large as needed to   |
+   |         | represent the given value faithfully.                    |
```
Comment (picnixz) [inline Doc/library/string.rst]: It's a hassle to make suggestions in RST tables so I'll write the text here:

"past the decimal point, and switches to the scientific notation when ``exp >= p - 1``. When the precision is not specified, the latter will be as large as needed to represent the given value faithfully."

Comment (picnixz) [reply Doc/library/string.rst]: (Sorry for reviewing it soooo late).

Comment (skirpichev) [reply Doc/library/string.rst]: Not a problem.

My major task (to figure out how it really works) was accomplished; the mpmath code for new-style formatting seems to be compatible with CPython.

**Review by picnixz — APPROVED** [review]

**Review by vstinner — APPROVED** [review]
LGTM.

I can see differences:

```
>>> format(1.0, '')  # at least one digit past dot
'1.0'
>>> format(1.0, 'g')
'1'

>>> format(1.2345e10, '')
'12345000000.0'
>>> format(1.2345e10, 'g')
'1.2345e+10'
```

**Comment by vstinner** [comment]
Merged. Thanks @skirpichev and @picnixz.

**Pull request merged**